internal/vsp/vsp.go

```diff
@@ -241,3 +241,38 @@ func (c *Client) ProcessWithPolicy(ctx context.Context, ticketHash *chainhash.Ha
 	}
 	return fp.submitPayment()
 }
+
+// SetVoteChoice takes all
+func (c *Client) SetVoteChoice(ctx context.Context, hash *chainhash.Hash, choices ...wallet.AgendaChoice) error {
+	c.mu.Lock()
+	_, ok := c.jobs[*hash]
+	c.mu.Unlock()
+	if ok {
+		// Already processing this ticket with the VSP.
+		return nil
+	}
```

Comment:
This will prevent us from setting vote choices for any ticket being processed.  I believe you meant to test for !ok.